commons-numbers-arrays/src/test/java/org/apache/commons/numbers/arrays/NormsTest.java

```diff
@@ -0,0 +1,564 @@
+
+        // act/assert
+        checkEuclideanRandom(2, rng, v -> Norms.euclidean(v[0], v[1]));
+    }
+
+    @Test
+    void testEuclidean_2d_vsArray() {
+        // arrange
+        final double[][] inputs = {
+            {-4.074598908124454E-9, 9.897869969944898E-28},
+            {1.3472131556526359E-27, -9.064577177323565E9},
+            {-3.9219339341360245E149, -7.132522817112096E148},
+            {-1.4888098520466735E153, -2.9099184907796666E150},
+            {-8.659395144898396E-152, -1.123275532302136E-150},
+            {-3.660198254902351E-152, -6.656524053354807E-153}
+        };
+
+        // act/assert
+        for (final double[] input : inputs) {
+            Assertions.assertEquals(Norms.euclidean(input), Norms.euclidean(input[0], input[1]),
+                () -> "Expected inline method result to equal array result for input " + Arrays.toString(input));
+        }
+    }
+
+    @Test
+    void testEuclidean_2d_vsHypot() {
+        // arrange
+        final int samples = 1000;
+        final UniformRandomProvider rng = RandomSource.create(RandomSource.XO_RO_SHI_RO_1024_PP, 2L);
+
+        // act/assert
+        assertEuclidean2dVersusHypot(-10, +10, samples, rng);
+        assertEuclidean2dVersusHypot(0, +20, samples, rng);
+        assertEuclidean2dVersusHypot(-20, 0, samples, rng);
+        assertEuclidean2dVersusHypot(-20, +20, samples, rng);
+        assertEuclidean2dVersusHypot(-100, +100, samples, rng);
+        assertEuclidean2dVersusHypot(+490, +510, samples, rng);
+        assertEuclidean2dVersusHypot(-510, -490, samples, rng);
+        assertEuclidean2dVersusHypot(-600, +600, samples, rng);
+    }
+
+    /** Assert that the Norms euclidean 2D computation produces similar error behavior to Math.hypot().
+     * @param minExp minimum exponent for random inputs
+     * @param maxExp maximum exponent for random inputs
+     * @param samples sample count
+     * @param rng random number generator
+     */
+    private static void assertEuclidean2dVersusHypot(final int minExp, final int maxExp, final int samples,
+            final UniformRandomProvider rng) {
+        // generate random inputs
+        final double[][] inputs = new double[samples][];
+        for (int i = 0; i < samples; ++i) {
+            inputs[i] = DoubleTestUtils.randomArray(2, minExp, maxExp, rng);
+        }
+
+        // compute exact results
+        final double[] exactResults = new double[samples];
+        for (int i = 0; i < samples; ++i) {
+            exactResults[i] = exactEuclideanNorm(inputs[i]);
+        }
+
+        // compute the std devs
+        final double hypotResult = computeUlpErrorStdDev(inputs, exactResults, v -> Math.hypot(v[0], v[1]));
+        final double normResult = computeUlpErrorStdDev(inputs, exactResults, v -> Norms.euclidean(v[0], v[1]));
+
+        // ensure that we are within the ballpark of Math.hypot
+        Assertions.assertTrue(normResult <= (hypotResult + HYPOT_COMPARE_EPS),
```

Comment:
You should also check the mean. What if Norms computed the exact result but shifted by 20 ULP. It would have a standard deviation of zero but a mean of 20. It would be very precise in its inaccuracy. Effectively you should test accuracy (mean ulp) and precision (std dev of the ulp).